### ISSUE TITLE
Renaming Object by clicking on object name in the left panel does not reflect the name change everywhere

### ISSUE THREAD

**Issue by panditthecoder** [objecthub-io/ObjectHub#245]
Another issue - usability

I clicked on the object name in the left panel (not in the browser section) and edited the name. clicked on the 'tick' mark to accept the change. the name changed in that panel but did not change in browser panel (see image)
![change_object_name_by_editing_name_link](https://user-images.githubusercontent.com/29138377/27546247-e7523232-5aaf-11e7-8579-285356eb0e42.JPG)

Logged out and back in and the changes were showing correctly. (see image).  
![correct_view_signout_sign_in_change_object_name_by_editing_name_link](https://user-images.githubusercontent.com/29138377/27546261-f2a873da-5aaf-11e7-8ed1-0a646569ec9f.JPG)

realized that i had not checked the recent tab. wet there and it was showing the correct name. however, that was because I had logged out and back in. so went back to the 'my objects' tab, clicked on the object to bring it to the left pane for editing, and changes the name again - as expected, the name change did not reflect in the 'my Objects' list but did reflect in the 'recent tab when i clicked on it. (see image)
![object_name_change_not_reflected_here_but_is_reflected_in_recent_tab_see_next_image](https://user-images.githubusercontent.com/29138377/27546394-46762714-5ab0-11e7-9c05-d616271cfe45.JPG)

and lastly, went back to the my Objects tab to ensure that moving between tabs did not refresh the data but it yet showed the old name (see image)
![next_image_as_promised](https://user-images.githubusercontent.com/29138377/27546451-6cb9d876-5ab0-11e7-9ec0-f46e7ef8b475.JPG)

IMHO, allowing users multiple ways to achieve the same objective (in this case renaming) is a recipe for disaster unless the objects are encapsulated perfectly and changes in one object trigger all related changes. Why do you need the ability to rename the object by clicking on the name - and if it is so, why not just call the same routine (not sure if it is being done or not) for rename in the browser ?

After typing the previous sentence, just for kicks, I went to the browser section to check if the rename was working correctly. clicked on the same object and clicked the 'Rename' icon. renames the object and as expected, it changed the name of the object in the 'My Objects; list as well as in the 'edit code' window. However, when i clicked on the 'recent' tab, the old name yet showed up. (see image)
![rename_from_browser_works_for_my_objects_but_not_for_recent](https://user-images.githubusercontent.com/29138377/27546853-a0c34c96-5ab1-11e7-9de4-4a9264272766.JPG)

so in a nutshell.
- if you rename from the code edit panel, it does not update the name in 'My Objects' - seems to do it for the 'recent' tab
- if you rename from the Browser panel, it updates the name in 'My Objects' - but Not for the 'recent' tab

I have not check the 'pin' functionality - suggest you check that as well while fixing it.

**Comment by objecthub-io** [comment]
Thank you for reporting this issue.

We could confirm it and will fix it.

**Comment by objecthub-io** [comment]
This should now be fixed. Renaming from object screen and browser should update in the respective other place.

**Comment by panditthecoder** [comment]
works as expected. can be closed

**Comment by objecthub-io** [comment]
Thank you for testing again - closing this issue now.
